compiler/rustc_builtin_macros/src/deriving/generic/mod.rs

```diff
@@ -1034,24 +1027,33 @@ impl<'a> MethodDef<'a> {
         self_args: &[P<Expr>],
         nonself_args: &[P<Expr>],
         use_temporaries: bool,
```

Comment:
I'm unsure exactly when `use_temporaries` is needed now that `is_packed` seems to be making many of the relevant decisions.  I tried to follow it down a bit more, but got a bit lost.

Are there any cases that strictly *need* temporaries when `is_packed` is false?

(I don't think this needs to block this PR, but might be worth exploring in future cleanup, especially if it could make callsites clearer to have an enum for the cases, rather than the bunch-of-`bool`s that some of them seem to have today.)

Reply:
`use_temporaries` is still used in `create_struct_pattern`, which has two call sites, so the `is_packed`/`use_temporaries` distinction is still required. I haven't changed this in any of my follow-ups, but I'll think about if it can be improved.